cmd/cmdutils.go

```diff
@@ -89,3 +91,107 @@ func validateName(name string) error {
+	fmt.Fprintln(tabWriterUnmounted, "NAME", "\t", "SIZE")
+
+	//iterating over all storages and put in the respective mount and unmount storage table
+	for _, storage := range storageList {
+		if storage.Path != "" {
+			if !hasMounted {
+				hasMounted = true
+			}
+			fmt.Fprintln(tabWriterMounted, storage.Name, "\t", storage.Size, "\t", storage.Path)
+		} else {
+			if !hasUnmounted {
+				hasUnmounted = true
+			}
+			fmt.Fprintln(tabWriterUnmounted, storage.Name, "\t", storage.Size)
+		}
+	}
+	//print all mounted storages of the given component
+	if hasMounted {
+		fmt.Printf("The component '%v' has the following storage attached:\n", componentName)
+		tabWriterMounted.Flush()
+	} else {
+		fmt.Printf("The component '%v' has no storage attached\n", componentName)
+	}
+	//print all unmounted storages of the given component
+	if hasUnmounted {
+		fmt.Printf("\nThe following unmounted storages can be mounted to '%v':\n", componentName)
+		tabWriterUnmounted.Flush()
+	} else {
+		fmt.Printf("\nNo unmounted storage exists to mount to '%v' \n", componentName)
+	}
+}
+
+// printStoragesInAllComponent prints all the storages in all the components of the application and project
+func printStoragesInAllComponent(client *occlient.Client, applicationName string, projectName string) {
+	componentList, err := component.List(client, applicationName, projectName)
+	checkError(err, "Could not get component list")
+
+	//this flag is initially false, will be true if we find any unmounted storage
+	hasUnmounted := false
+	//defining the column structure of the table
+	tabWriterUnmounted := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+	//create header of unmounted storages in all the components of the given application and project
+	fmt.Fprintln(tabWriterUnmounted, "NAME", "\t", "SIZE")
+
+	//iterating over all the components in the given aplication and project
+	for _, component := range componentList {
+		componentName := component.Name
+		storageList, err := storage.List(client, componentName, applicationName)
+		checkError(err, "")
+
+		//this flag is initially false, will be true if we find any mounted storage in the given component
+		hasMounted := false
+		//defining the column structure of the table
+		tabWriterMounted := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+
+		//create headers of mounted storages in the current component of the given application and project
+		fmt.Fprintln(tabWriterMounted, "NAME", "\t", "SIZE", "\t", "PATH")
+
+		//iterating over all storages and put in the respective mount and unmount storage table
+		for _, storage := range storageList {
+			if storage.Path != "" {
+				if !hasMounted {
+					hasMounted = true
+				}
+				fmt.Fprintln(tabWriterMounted, storage.Name, "\t", storage.Size, "\t", storage.Path)
+			} else {
+				if !hasUnmounted {
+					hasUnmounted = true
+				}
+				fmt.Fprintln(tabWriterUnmounted, storage.Name, "\t", storage.Size)
```

Comment:
Storage.List() returns the unmounted storage in the same application. So if there are two or more components in the same application and one or more unmounted storage in the application, then this will display the same unmounted storage more than once as the unmounted storage are getting written to the tabWriterUnmounted more than once 
```
[mdas@localhost odo]$ ./odo storage list --all
The component 'nodejs' has the following storage attached -
NAME               SIZE     PATH
backend-nodejs     100M     /data-1

The component 'wildfly' has no storage attached

Storages that are not mounted to any component -
NAME             SIZE
backend          100M
backend-wild     100M
backend          100M
backend-wild     100M
```